packages/boilerplates/BoilerplatesLoader.ts

```diff
@@ -0,0 +1,184 @@
+        let boilerplatePath = path.join(folder, 'boilerplate.json');
+        
+        if (!this._fileSystem.existsSync(boilerplatePath)) {
+            this._logger.info(`The path of a boilerplate defined in the boilerplates configuration does not exists. Path: ${boilerplatePath}`);
+            throw new Error(`Could not find boilerplate configuration in '${folder}'`);
+        }
+
+        let boilerplateObject = JSON.parse(this._fileSystem.readFileSync(boilerplatePath, 'utf8'));
+
+        return this.parseBaseBoilerplate(boilerplateObject, boilerplatePath);
+    }
+
+    private parseBaseBoilerplate(boilerplateObject: any, boilerplatePath: string): BaseBoilerplate {
+        if (boilerplateObject.type === artifactsBoilerplateType) 
+            return this.parseArtifactsBoilerplate(boilerplateObject, boilerplatePath);
+        else 
+            return this.parseBoilerplate(boilerplateObject, boilerplatePath);
+    }
+
+    private parseArtifactsBoilerplate(boilerplateObject: any, boilerplatePath: string): ArtifactsBoilerplate {
+        return new ArtifactsBoilerplate(
+            boilerplateObject.language || 'any',
+            boilerplateObject.name,
+            boilerplateObject.description,
+            boilerplateObject.type,
+            boilerplateObject.dependencies !== undefined? 
+                Object.keys(boilerplateObject.dependencies).map(key => dependencyParsers.parse(boilerplateObject.dependencies[key], key))
+                : [],
+            boilerplateObject.namespace,
+            Scripts.fromJson(boilerplateObject.scripts),
+            boilerplatePath,
+            this._folders,
+            this._fileSystem);
+    }
+    private parseBoilerplate(boilerplateObject: any, boilerplatePath: string): Boilerplate {
+        let bindings = this.getBoilerplateBindings(boilerplatePath);
+        return new Boilerplate(
+            boilerplateObject.language || 'any',
+            boilerplateObject.name,
+            boilerplateObject.description,
+            boilerplateObject.type,
+            boilerplateObject.dependencies !== undefined? 
+                Object.keys(boilerplateObject.dependencies).map(key => dependencyParsers.parse(boilerplateObject.dependencies[key], key))
+                : [],
+            boilerplateObject.namespace,
+            Scripts.fromJson(boilerplateObject.scripts),
+            boilerplateObject.target,
+            boilerplateObject.framework,
+            boilerplateObject.parent,
+            boilerplatePath,
+            bindings.pathsNeedingBinding,
+            bindings.filesNeedingBinding
+        );
+    }
+    /**
+     * Gets the path and file bindings for a boilerplate
+     * 
+     * @param {string} boilerplatePath The path to the boilerplate.json file
+     * @returns {{pathsNeedingBinding: string[], filesNeedingBinding: string[]}} 
+     */
+    private getBoilerplateBindings(boilerplatePath: string): { pathsNeedingBinding: string[]; filesNeedingBinding: string[]; } {
```

Comment:
better name; `getBindingsFor()`
No need for prefixing in the `boilerplatePath`- just path.